### PR TITLE
open-watcom-v2-unwrapped: unstable-2022-10-03 -> unstable-2023-01-30, more platforms

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Description of changes

- bump
- adjust patching to some changes to the date embedding
- enable on more platforms
  - [Aarch64](https://github.com/open-watcom/open-watcom-v2/blob/6609ddae017214cf702b8ed541a0d378abcb16ca/build/makeinit#L111) (new, tested on aarch64-linux, aarch64-darwin untested)
  - [32-bit ARM](https://github.com/open-watcom/open-watcom-v2/blob/6609ddae017214cf702b8ed541a0d378abcb16ca/build/makeinit#L107) (already existed but I can't test it)
  - [MIPS](https://github.com/open-watcom/open-watcom-v2/blob/6609ddae017214cf702b8ed541a0d378abcb16ca/build/makeinit#L101) (32-bit support should be there but the wrapper won't know where the binaries are placed, feedback from anyone on this would be appreciated!)

(PPC should be supported too but there's no convenient entry for that in doubles.nix, wrapper also won't know where the binaries are)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OPNA2608** [comment]
@ofborg build open-watcom-v2-unwrapped

**Comment by OPNA2608** [comment]
Hmm, seems like DOSBox doesn't build on aarch64-darwin… Gonna have to test this another time I guess.

**Comment by OPNA2608** [comment]
Looks like DOSBox got fixed on aarch64-darwin, maybe worth another test?

**Comment by OPNA2608** [comment]
@ofborg build open-watcom-v2-unwrapped

**Comment by OPNA2608** [comment]
Hmm still having no luck with the aarch64-darwin CI, this time for reasons that feel entirely outside of my control. Must not be my year haha.

```
deleting unused links...
note: currently hard linking saves 0.00 MiB
copying path '/nix/store/b6y2prxms56m2kzssqdkcgxxrf83yx4d-clang-11.1.0' from 'https://cache.nixos.org'...
trying https://github.com/open-watcom/open-watcom-v2/archive/de461d35171d88ea80030608597699cb89e1dbe6.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
copying path '/nix/store/dkmq62ibx58cmci1nq1rbwsc0y067r85-cctools-binutils-darwin-973.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/kvqy6garv73hnn6iiaparwicvskacflp-cctools-binutils-darwin-wrapper-973.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/fb4rdxr69sy1vsi98z3njxjzdq3rj3ah-clang-wrapper-11.1.0' from 'https://cache.nixos.org'...
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
copying path '/nix/store/mnr2a6hn5ng7a0rdnklwlk3k8ifpvvi5-stdenv-darwin' from 'https://cache.nixos.org'...
running auto-GC to free 52286640128 bytes
finding garbage collector roots...
deleting garbage...
deleting unused links...
note: currently hard linking saves 0.00 MiB
100 69.9M    0 69.9M    0     0  6974k      0 --:--:--  0:00:10 --:--:-- 9375k
unpacking source archive /private/tmp/nix-build-source.drv-0/de461d35171d88ea80030608597699cb89e1dbe6.tar.gz
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/bld/clib/startup/a/seg16.asm': No space left on device
running auto-GC to free 53642264576 bytes
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/bld/clib/startup/a/seg386.asm': No space left on device
error (ignored): error: cannot bind to socket '/nix/var/nix/gc-socket/socket': No space left on device
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/bld/clib/startup/a/seh386v2.asm': No space left on device
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/bld/clib/startup/a/stjmp086.asm': No space left on device
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/bld/clib/startup/a/stjmp386.asm': No space left on device
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/bld/clib/startup/a/stjmpaxp.asm': No space left on device
[...]
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/zipup.sh': No space left on device
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/zipuprel.cmd': No space left on device
mv: cannot create regular file '/nix/store/mkmlzcqh93w8f1hbrvm231ixpk397ahx-source/zipuprel.sh': No space left on device
error: writing to file: No space left on device
```

File: pkgs/development/compilers/open-watcom/v2.nix [hunk 1]
```diff
@@ -120,7 +120,7 @@ stdenv.mkDerivation rec {
     '';
     homepage = "https://open-watcom.github.io";
     license = licenses.watcom;
-    platforms = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "x86_64-windows" "i686-windows" ];
+    platforms = with platforms; arm ++ aarch64 ++ x86 ++ mips; # ++ powerpc
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/open-watcom/v2.nix:123]:
```suggestion
    platforms = platforms.unix;
```

Comment (OPNA2608) [reply pkgs/development/compilers/open-watcom/v2.nix:123]: No, just Unix in general really isn't right… If this is too specific for you then I'll compromise by just marking some known-incompatible platforms in `badPlatforms`.

File: pkgs/development/compilers/open-watcom/v2.nix [hunk 2]
```diff
@@ -120,7 +123,8 @@ stdenv.mkDerivation rec {
     '';
     homepage = "https://open-watcom.github.io";
     license = licenses.watcom;
-    platforms = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "x86_64-windows" "i686-windows" ];
+    platforms = with platforms; windows ++ unix;
+    badPlatforms = platforms.riscv ++ [ "powerpc64-linux" "powerpc64le-linux" "mips64el-linux" ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/open-watcom/v2.nix:127]: We don't have any hydra builders for those platforms and they have a rather low support tier, so I wouldn't be surprised if not many of the dependencies already fail to build.

**Pull request merged**